### PR TITLE
Improved script copying to prevent unhelpful errors

### PR DESCRIPTION
**Pull request by bitcrazed**
* Moved common script copying code to new `safe-filecopy` function
* Added additional destination file already exists check with graceful fallback
* Added a little color pizazz - a useful visual cue

Fixes issue bitcrazed/dnx#1424

### CODE REVIEW

**Comment by msotclas** [comment]
Hi __@bitcrazed__, I'm your friendly neighborhood Microsoft Open Technologies, Inc. Pull Request Bot (You can call me MSOTBOT). Thanks for your contribution!
    <span>
        In order for us to evaluate and accept your PR, we ask that you sign a contribution license agreement. It's all electronic and will take just minutes. I promise there's no faxing. https://cla2.msopentech.com.
    </span>

TTYL, MSOTBOT;

**Comment by msotclas** [comment]
__@bitcrazed__, Thanks for signing the contribution license agreement so quickly! Actual humans will now validate the agreement and then evaluate the PR.
<br />Thanks, MSOTBOT;

**Comment by glennc** [comment]
:shipit: @anurse anything you want to add?

**Comment by bitcrazed** [comment]
@anurse & @glennc: Let me know if there's anything else you need done to this PR

File: src/dnvm.ps1 [hunk 1]
```diff
@@ -1173,6 +1173,51 @@ function dnvm-name {
     Get-RuntimeName $VersionOrAlias $Architecture $Runtime
 }
 
+
+<#
+.SYNOPSIS
+    Checks if a specified file exists in the destination folder and if not, copies the file
+    to the destination folder. 
+.PARAMETER Filaneme
```
Comment (anurse) [inline src/dnvm.ps1]: Nit: Typo :)

File: src/dnvm.ps1 [hunk 2]
```diff
@@ -1173,6 +1173,51 @@ function dnvm-name {
...
+    Name of the file to copy
+.PARAMETER SourceFolder
+    Path to the folder containing the source file
+.PARAMETER DestinationFolder
+    Path to the folder into which the source file should be copied
+#>
+function safe-filecopy {
```
Comment (anurse) [inline src/dnvm.ps1]: Nit: Title-Case please :)

File: src/dnvm.ps1 [hunk 3]
```diff
@@ -1173,6 +1173,51 @@ function dnvm-name {
     Get-RuntimeName $VersionOrAlias $Architecture $Runtime
 }
 
+
+<#
```
Comment (anurse) [inline src/dnvm.ps1]: Internal functions don't really need the full PowerShell Docs treatment, but hey, I won't complain if you put it in :D

File: src/dnvm.ps1 [hunk 4]
```diff
@@ -1173,6 +1173,51 @@ function dnvm-name {
...
+
+    if(Test-Path $sourceFilePath) {
+        _WriteOut -NoNewLine "Installing '$Filename' to '$DestinationFolder' ... "
+
+        if (Test-Path $destFilePath) {
+            _WriteOut "Skipping: file already exists" -ForegroundColor Yellow
+        }
+        else {
+            Copy-Item $sourceFilePath $destFilePath -Force
+            _WriteOut "Done" -ForegroundColor "Green"
```
Comment (anurse) [inline src/dnvm.ps1]: Copies are pretty fast, I wouldn't worry about writing the "Done" message.

File: src/dnvm.ps1 [hunk 5]
```diff
@@ -1173,6 +1173,51 @@ function dnvm-name {
...
+        }
+    }
+    else {
+        _WriteOut "WARNING: Unable to install: Could not find '$Filename' in '$SourceFolder'. " }
```
Comment (anurse) [inline src/dnvm.ps1]: Nit: Closing "}" on new line please :)

**Comment by anurse** [comment]
Looks great! Thanks so much! A couple small nits to be picked but otherwise I think it's a great change!

File: src/dnvm.ps1 (hunk 1 above)
Comment (bitcrazed) [inline src/dnvm.ps1]: Thanks. Fixed :eyeglasses:

File: src/dnvm.ps1 (hunk 2 above)
Comment (bitcrazed) [inline src/dnvm.ps1]: Fixed casing. Removed doc comments, left rationale comment behind for future reference.

As a matter of interest, why Title-Case for internal functions, but lower-case for public (e.g. `dnvm-*`)?

File: src/dnvm.ps1 (hunk 3 above)
Comment (bitcrazed) [inline src/dnvm.ps1]: :laughing: Removed doc comments, left rationale comment behind for future reference.

File: src/dnvm.ps1 (hunk 4 above)
Comment (bitcrazed) [inline src/dnvm.ps1]: ... Done! :stuck_out_tongue_winking_eye:

File: src/dnvm.ps1 (hunk 5 above)
Comment (bitcrazed) [inline src/dnvm.ps1]: Nice catch - COMPLETELY agree!

**Comment by bitcrazed** [comment]
Thanks for the review comments. Requested changes made.

**Comment by anurse** [comment]
Looks good! :shipit:

**Comment by bitcrazed** [comment]
WOOT! Thanks 😁 Achievement Unlocked 😏

**Pull request merged**